src/Hl7.Fhir.Core/Rest/Http/EntryToHttpExtensions.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * Copyright (c) 2014, Furore (info@furore.com) and contributors
+ * See the file CONTRIBUTORS for details.
+ *
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/ewoutkramer/fhir-net-api/master/LICENSE
+ */
+
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Hl7.Fhir.Utility;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Linq;
+
+namespace Hl7.Fhir.Rest.Http
+{
+    internal static class EntryToHttpExtensions
+    {
+        public static HttpRequestMessage ToHttpRequestMessage(this Bundle.EntryComponent entry, 
+            SearchParameterHandling? handlingPreference, Prefer? returnPreference, ResourceFormat format, bool useFormatParameter, bool CompressRequestBody)
+        {
+            System.Diagnostics.Debug.WriteLine("{0}: {1}", entry.Request.Method, entry.Request.Url);
+
+            var interaction = entry.Request;
+
+            if (entry.Resource != null && !(interaction.Method == Bundle.HTTPVerb.POST || interaction.Method == Bundle.HTTPVerb.PUT))
+                throw Error.InvalidOperation("Cannot have a body on an Http " + interaction.Method.ToString());
+
+            var location = new RestUrl(interaction.Url);
+
+            if (useFormatParameter)
+                location.AddParam(HttpUtil.RESTPARAM_FORMAT, Hl7.Fhir.Rest.ContentType.BuildFormatParam(format));
+
+            var request = new HttpRequestMessage(getMethod(interaction.Method), location.Uri);
+
+            if (!useFormatParameter)
+                request.Headers.Add("Accept", Hl7.Fhir.Rest.ContentType.BuildContentType(format, forBundle: false));
```

Comment:
What is the reason you are not doing request.Headers.Accept.Add()?

Reply:
Did not see static `MediaTypeWithQualityHeaderValue.Parse`. Replaced this with a call to  `request.Headers.Accept.Add`.